### PR TITLE
feat: Add slow flags for commands with heavy disk I/O

### PR DESCRIPTION
**Pull request by jonathanc-n**
I have added slow flags for the following commands:
- HGETALL
- LINSERT
- LREM
- LRANGE
- SCAN
- SINTER
- SDIFF
- SUNION

I would like to just see if I had done these correctly, I am looking to add support for these commands as well (please verify if these seem to be commands that contribute to heavy disk I/O + lmk if there are more to add):
- MSET
- MGET
- MSETNX
- GETRANGE
- SETRANGE
- HKEYS
- HVALS
- HMGET
- HRANGEBYLEX
- LMPOP
- BLMPOP
- SCARD
- SMEMBERS
- ZRANGEBYSCORE
- ZRANGEBYLEX
- ZUNION
- ZINTER
- SORT

References: #2474

### CODE REVIEW

**Comment by git-hulk** [comment]
@jonathanc-n What this issue intends to add a command attribute like what we have in [commands/commander.h#L66](https://github.com/apache/kvrocks/blob/unstable/src/commands/commander.h#L66).

cc @PragmaTwice 


And for the slow commands:

- SCAN is not a slow command
- HKEYS/HVALS are the same as HGETALL, and can be identified as an O(N) command
- ZUNION/ZINTER are the same as SUNION
- SMEMBERS is a slow command
- *BYLEX commands behave like the SCAN command, so it shouldn't be a slow command

**Comment by jonathanc-n** [comment]
@git-hulk Alright got it, thanks for the feedback!

**Comment by jonathanc-n** [comment]
@git-hulk Are you able to check if this looks good? Thank you in advance!

File: src/commands/cmd_hash.cc [hunk 1]
```diff
@@ -299,7 +299,6 @@ class CommandHGetAll : public Commander {
     if (!s.ok()) {
       return {Status::RedisExecErr, s.ToString()};
     }
-
```
Comment (git-hulk) [inline src/commands/cmd_hash.cc]: Can reverse this line.

File: src/commands/cmd_hash.cc [hunk 2]
```diff
@@ -440,11 +439,11 @@ REDIS_REGISTER_COMMANDS(Hash, MakeCmdAttr<CommandHGet>("hget", 3, "read-only", 1
                         MakeCmdAttr<CommandHLen>("hlen", 2, "read-only", 1, 1, 1),
                         MakeCmdAttr<CommandHMGet>("hmget", -3, "read-only", 1, 1, 1),
                         MakeCmdAttr<CommandHMSet>("hmset", -4, "write", 1, 1, 1),
-                        MakeCmdAttr<CommandHKeys>("hkeys", 2, "read-only", 1, 1, 1),
-                        MakeCmdAttr<CommandHVals>("hvals", 2, "read-only", 1, 1, 1),
+                        MakeCmdAttr<CommandHKeys>("hkeys", 2, "read-only slow", 1, 1, 1),
+                        MakeCmdAttr<CommandHVals>("hvals", 2, "read-only slow", 1, 1, 1),
                         MakeCmdAttr<CommandHGetAll>("hgetall", 2, "read-only", 1, 1, 1),
```
Comment (git-hulk) [inline src/commands/cmd_hash.cc]: HGETALL should be a slow command

File: src/commands/cmd_hash.cc [hunk 3]
```diff
@@ -440,11 +439,11 @@ REDIS_REGISTER_COMMANDS(Hash, MakeCmdAttr<CommandHGet>("hget", 3, "read-only", 1
...
                         MakeCmdAttr<CommandHScan>("hscan", -3, "read-only", 1, 1, 1),
-                        MakeCmdAttr<CommandHRangeByLex>("hrangebylex", -4, "read-only", 1, 1, 1),
+                        MakeCmdAttr<CommandHRangeByLex>("hrangebylex", -4, "read-only slow", 1, 1, 1),
```
Comment (git-hulk) [inline src/commands/cmd_hash.cc]: HRANGEBYLET is not a slow command

File: src/commands/cmd_list.cc [hunk 4]
```diff
@@ -633,7 +633,6 @@ class CommandLTrim : public Commander {
 
     start_ = *parse_start;
     stop_ = *parse_stop;
-
```
Comment (git-hulk) [inline src/commands/cmd_list.cc]: Can revert this line.

File: src/commands/cmd_server.cc [hunk 5]
```diff
@@ -838,7 +838,6 @@ class CommandScan : public CommandScanBase {
   Status Execute(Server *srv, Connection *conn, std::string *output) override {
     redis::Database redis_db(srv->storage, conn->GetNamespace());
     auto key_name = srv->GetKeyNameFromCursor(cursor_, CursorType::kTypeBase);
-
```
Comment (git-hulk) [inline src/commands/cmd_server.cc]: Can revert this line.

File: src/commands/cmd_set.cc [hunk 6]
```diff
@@ -440,15 +440,15 @@ class CommandSScan : public CommandSubkeyScanBase {
...
                         MakeCmdAttr<CommandSPop>("spop", -2, "write", 1, 1, 1),
                         MakeCmdAttr<CommandSRandMember>("srandmember", -2, "read-only", 1, 1, 1),
                         MakeCmdAttr<CommandSMove>("smove", 4, "write", 1, 2, 1),
-                        MakeCmdAttr<CommandSDiff>("sdiff", -2, "read-only", 1, -1, 1),
-                        MakeCmdAttr<CommandSUnion>("sunion", -2, "read-only", 1, -1, 1),
-                        MakeCmdAttr<CommandSInter>("sinter", -2, "read-only", 1, -1, 1),
+                        MakeCmdAttr<CommandSDiff>("sdiff", -2, "read-only slow", 1, -1, 1),
+                        MakeCmdAttr<CommandSUnion>("sunion", -2, "read-only slow", 1, -1, 1),
+                        MakeCmdAttr<CommandSInter>("sinter", -2, "read-only slow", 1, -1, 1),
                         MakeCmdAttr<CommandSInterCard>("sintercard", -3, "read-only", CommandSInterCard::Range),
```
Comment (git-hulk) [inline src/commands/cmd_set.cc]: sintercard/sdiffstore/sunionstore/sinterstore should be slow commands as well.

File: src/commands/cmd_zset.cc [hunk 7]
```diff
@@ -1517,7 +1517,7 @@ REDIS_REGISTER_COMMANDS(ZSet, MakeCmdAttr<CommandZAdd>("zadd", -4, "write", 1, 1
                         MakeCmdAttr<CommandZCount>("zcount", 4, "read-only", 1, 1, 1),
                         MakeCmdAttr<CommandZIncrBy>("zincrby", 4, "write", 1, 1, 1),
                         MakeCmdAttr<CommandZInterStore>("zinterstore", -4, "write", CommandZInterStore::Range),
-                        MakeCmdAttr<CommandZInter>("zinter", -3, "read-only", CommandZInter::Range),
+                        MakeCmdAttr<CommandZInter>("zinter", -3, "read-only slow", CommandZInter::Range),
                         MakeCmdAttr<CommandZInterCard>("zintercard", -3, "read-only", CommandZInterCard::Range),
```
Comment (git-hulk) [inline src/commands/cmd_zset.cc]: Same as the SET commands, ZINTER*/ZUNION*/ZDIFF* should be identified as a slow command.

File: src/server/redis_connection.cc [hunk 8]
```diff
@@ -367,6 +367,7 @@ static bool IsCmdForIndexing(const CommandAttributes *attr) {
           attr->category == CommandCategory::Key);
 }
 
+// TODO: Handle slow flag
```
Comment (git-hulk) [inline src/server/redis_connection.cc]:
```suggestion
```

**Comment by jonathanc-n** [comment]
@git-hulk Thanks for the review, sorry about the messy PR. I will make sure it is better in the future.

File: src/commands/cmd_list.cc [hunk 9]
```diff
@@ -852,18 +852,18 @@ REDIS_REGISTER_COMMANDS(List, MakeCmdAttr<CommandBLPop>("blpop", -3, "write no-s
...
                         MakeCmdAttr<CommandLPos>("lpos", -3, "read-only", 1, 1, 1),
                         MakeCmdAttr<CommandLPush>("lpush", -3, "write", 1, 1, 1),
                         MakeCmdAttr<CommandLPushX>("lpushx", -3, "write", 1, 1, 1),
-                        MakeCmdAttr<CommandLRange>("lrange", 4, "read-only", 1, 1, 1),
-                        MakeCmdAttr<CommandLRem>("lrem", 4, "write no-dbsize-check", 1, 1, 1),
+                        MakeCmdAttr<CommandLRange>("lrange", 4, "read-only slow", 1, 1, 1),
+                        MakeCmdAttr<CommandLRem>("lrem", 4, "write no-dbsize-check slow", 1, 1, 1),
                         MakeCmdAttr<CommandLSet>("lset", 4, "write", 1, 1, 1),
-                        MakeCmdAttr<CommandLTrim>("ltrim", 4, "write no-dbsize-check", 1, 1, 1),
+                        MakeCmdAttr<CommandLTrim>("ltrim", -4, "write no-dbsize-check", 1, 1, 1),
```
Comment (mapleFU) [inline src/commands/cmd_list.cc]: why changing to -4?

Comment (jonathanc-n) [reply src/commands/cmd_list.cc]: Oh sorry, I didn't see that, it was a change to the past arity. Should be a positive

File: src/commands/cmd_list.cc [hunk 10]
```diff
@@ -852,16 +852,16 @@ REDIS_REGISTER_COMMANDS(List, MakeCmdAttr<CommandBLPop>("blpop", -3, "write no-s
...
                         MakeCmdAttr<CommandLLen>("llen", 2, "read-only", 1, 1, 1),
                         MakeCmdAttr<CommandLMove>("lmove", 5, "write", 1, 2, 1),
                         MakeCmdAttr<CommandBLMove>("blmove", 6, "write", 1, 2, 1),
                         MakeCmdAttr<CommandLPop>("lpop", -2, "write", 1, 1, 1),  //
                         MakeCmdAttr<CommandLPos>("lpos", -3, "read-only", 1, 1, 1),
                         MakeCmdAttr<CommandLPush>("lpush", -3, "write", 1, 1, 1),
                         MakeCmdAttr<CommandLPushX>("lpushx", -3, "write", 1, 1, 1),
-                        MakeCmdAttr<CommandLRange>("lrange", 4, "read-only", 1, 1, 1),
-                        MakeCmdAttr<CommandLRem>("lrem", 4, "write no-dbsize-check", 1, 1, 1),
+                        MakeCmdAttr<CommandLRange>("lrange", 4, "read-only slow", 1, 1, 1),
```
Comment (git-hulk) [inline src/commands/cmd_list.cc]:
```suggestion
                        MakeCmdAttr<CommandLRange>("lrange", 4, "read-only", 1, 1, 1),
```

File: src/commands/cmd_list.cc [hunk 11]
```diff
@@ -495,6 +494,7 @@ class CommandLInsert : public Commander {
     } else {
       return {Status::RedisParseErr, errInvalidSyntax};
     }
+
```
Comment (git-hulk) [inline src/commands/cmd_list.cc]:
```suggestion
```

File: src/commands/cmd_list.cc [hunk 12]
```diff
@@ -464,7 +464,6 @@ class CommandLRem : public Commander {
     if (!parse_result) {
       return {Status::RedisParseErr, errValueNotInteger};
     }
-
```
Comment (git-hulk) [inline src/commands/cmd_list.cc]: Revert this line

**Comment by git-hulk** [comment]
@jonathanc-n A few new comments, rest are good to me. Thank you.

**Review by git-hulk — APPROVED** [review]
LGTM

**Review by mapleFU — APPROVED** [review]
I believe this is not enough, some command in kvrocks don't have a good implementation, so it's not slow in redis but slow in kvrocks

However, we can move forward firstly

**Pull request merged**